### PR TITLE
Refactor Return Types **BIG CHANGES**

### LINKED ISSUE #109: Refactor error returns
Problem
-------

Currently, error return formats are often left up to the internal methods in which they are generated. The errors are returned as a dict-in-a-list to the public method, then returned immediately to the user. This causes a problem with singular return methods that return a dict (**not** a dict-in-a-list) because the errors are returned in a list but successes are not. This causes the obvious problem of the user parsing the return.

Solution
--------

All internal methods return the error as a dict. Public methods then format the error to match success return format. This way, all error formats match their current method's success format.

### PR DESCRIPTION
**Pull request by matthewrsj**
Fixes #109
- `get_*` returns dict(s) within a list
- `create_*` returns a dict
- `update_*` returns a dict
- `authenticate()` returns a dict
- other simple data returns (that don't return multiple objects)
  also return a dict or single element (e.g. datetime)
- error message returns are formatted identically to their parent
  method

### CODE REVIEW

File: docs/source/usage.rst [hunk 1]
```diff
@@ -535,7 +546,7 @@ TimeSync.\ **get_users(username=None)**
...
+      [{u'username': u'userone', u'displayname': u'One Is The Loneliest Number', u'admin': False, u'created_at': u'2015-02-29', u'active': True, u'deleted_at': None, u'email': u'exampleone@example.com'}, {u'username': u'usertwo', u'displayname': u'Two Can Be As Bad As One', u'admin': False, u'created_at': u'2015-02-29', u'active': True, u'deleted_at': None, u'email': u'exampletwo@example.com'}, {u'username': u'userthree', u'displayname': "Yes It's The Saddest Experience", u'admin': False, u'created_at': u'2015-02-29', u'active': True, u'deleted_at': None, u'email': u'examplethree@example.com'}, {u'username': u'userfour', u'displayname': "You'll Ever Do", u'admin': False, u'created_at': u'2015-02-29', u'active': True, u'deleted_at': None, u'email': u'examplefour@example.com'}]
```
Comment (athai) [inline docs/source/usage.rst]: Display names for userthree and userfour are encased in double quotes. They're also missing the 'u' that's preceding all the strings.

**Comment by Kennric** [comment]
I don't see any issue with this. +1

File: docs/source/usage.rst [hunk 2]
```diff
@@ -585,7 +595,7 @@ TimeSync.\ **create_project(project)**
...
+      {u'users': {u'tschuy': {u'member': true, u'spectator': false, u'manager': true}, u'mrsj': {u'member': true, u'spectator': false, u'manager': false}, u'patcht': {u'member': true, u'spectator': false, u'manager': true}, u'oz': {u'member': false, u'spectator': true, u'manager': false}}, u'deleted_at': None, u'uuid': u'309eae69-21dc-4538-9fdc-e6892a9c4dd4', u'updated_at': None, u'created_at': u'2015-05-23', u'uri': u'https://code.osuosl.org/projects/timesync', u'name': u'TimeSync API', u'revision': 1, u'slugs': [u'timesync', u'time'], u'users': {u'managers': [u'tschuy'], u'spectators': [u'tschuy'], u'members': [u'patcht', u'tschuy']}}
```
Comment (athai) [inline docs/source/usage.rst]: Did you mean to add mrsj and oz to the list of users (and user perms) associated with the newly created project?

File: docs/source/usage.rst [hunk 3]
```diff
@@ -646,7 +656,7 @@ TimeSync.\ **delete_project(slug)**
     .. code-block:: python
 
       >>> ts.delete_project(slug="some-slug")
-      [{"status": 200}]
+      {u'status': 200}]
```
Comment (athai) [inline docs/source/usage.rst]: Square bracket at the end needs to be removed

**Comment by matthewrsj** [comment]
@athai good eye! They should be fixed now.

File: docs/source/usage.rst [hunk 4]
```diff
@@ -626,7 +636,7 @@ TimeSync.\ **update_project(project, slug)**
       ...    "name": "pymesync",
       ...}
       >>> ts.update_project(project=project, slug="ps")
-      [{'users': {'tschuy': {'member': true, 'spectator': true, 'manager': true}, 'patcht': {'member': true, 'spectator': false, 'manager': false}}, 'uuid': '309eae69-21dc-4538-9fdc-e6892a9c4dd4', 'name': 'pymesync', 'updated_at': '2014-04-18', 'created_at': '2014-04-16', 'deleted_at': None, 'revision': 2, 'uri': 'https://code.osuosl.org/projects/timesync', 'slugs': ['ps']}]
+      {u'users': {u'tschuy': {u'member': True, u'spectator': True, u'manager': True}, u'patcht': {u'member': True, u'spectator': False, u'manager': False}}, u'uuid': u'309eae69-21dc-4538-9fdc-e6892a9c4dd4', u'name': u'pymesync', u'updated_at': u'2014-04-18', u'created_at': u'2014-04-16', u'deleted_at': None, u'revision': 2, u'uri': u'https://code.osuosl.org/projects/timesync', u'slugs': [u'ps']}]
```
Comment (Kennric) [inline docs/source/usage.rst]: Looks like you have another trailing ] here

**Comment by Kennric** [comment]
Looks good to me. +1

**Comment by athai** [comment]
+1 

Although I'm curious, is 'oz' an abbreviation for something? It's listed as a user twice.

**Comment by matthewrsj** [comment]
Oz isn't an abbreviation. Sometimes I just make users based on what is going through my mind. Maybe I was thinking of the Wizard Of Oz for some reason.

**Pull request merged**